### PR TITLE
[FRONT-801] Finish edit member nickname modal

### PR DESCRIPTION
**Pull request by its-edmund**

### CODE REVIEW

File: src/components/RoomDropdown/index.tsx [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    background: #ffffff;
+    border-radius: 10px;
+    box-shadow: 0px 0px 10px rgba(0, 0, 0, 0.1);
+    box-sizing: border-box;
+    color: #59799c;
+    font-family: ${KARELIA};
+    font-size: 14px;
+    list-style-type: none;
+    border-spacing: 0px;
+    hr {
```
Comment (mondoreale) [inline src/components/RoomDropdown/index.tsx:81]: Let's separate selectors with an empty line.

File: src/components/RoomDropdown/index.tsx [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+        padding-bottom: 15px;
+        border-radius: 0 0 10px 10px;
+    }
+    :hover {
+        background-color: #f1f4f7;
+    }
+`
+
+const StyledModalContent = styled.div`
+    .contentContainer {
```
Comment (mondoreale) [inline src/components/RoomDropdown/index.tsx:112]: Those are real class names used in our code (in `RoomDropdown`). Let's turn them into styled components, too.

**Review by mondoreale — CHANGES_REQUESTED** [review]
Member list has a horizontal scrollbar.

Screen Shot 2022-01-05 at 2 01 11 PM

There's a bunch of other things here to iron out but we gotta get #11 merged.

Cheers!

File: src/components/RoomDropdown/index.tsx (hunk 2 above)
Comment (mxinburritos) [reply src/components/RoomDropdown/index.tsx:112]: Not that familiar with styled-components best practices but should I avoid using classnames and instead make all of the child components styled-components?

Comment (mondoreale) [reply src/components/RoomDropdown/index.tsx]: Yes*, please, for consistency. It'll also just appreciate in value over time (no need to keep an eye out for name conflicts and so on).

*) In practice not all child components have to be converted. Depends on the situation. Example

```tsx
const MyDivo = styled.div`
    color: red;

    h1 {
        color: blue;
    }
`

const div = (
    <MyDivo>
        <h1>Hello world</h1>
    </MyDivo>
)
```
If there's just one `h1` then there's no need for a separate class. If you can come up with a simple but deterministic selector for an element then there's no need for a class.

---

You may already know this… You can style styled-components within other styled-components. Example

```tsx
const A = styled.div``

const B = styled.div`
    ${A} {
        color: red;
    }
`
```

It's neat.

File: src/components/RoomDropdown/MemberOptions.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    padding: 7px 12px;
+`
+
+const UnstyledMemberOptions = ({ address }: any) => {
+    const [editing, setEditing] = useState(false)
+    const [nickname, setNickname] = useState('')
+    return (
+        <Root>
+            <MemberIcon />
+            <div style={{ display: 'flex', flexDirection: 'column' }}>
```
Comment (mondoreale) [inline src/components/RoomDropdown/MemberOptions.tsx:63]: Can we move `style` into css?

**Review by mondoreale — CHANGES_REQUESTED** [review]
Getting there! Let's merge #11 to clean up the noise. Then there's this little visual bug:

Screen Shot 2022-01-18 at 9 59 05 AM

LMK if you need help with this small glitch. For now I'm requesting changes.

File: src/components/RoomDropdown/index.tsx (hunk 2 above)
Comment (its-edmund) [reply src/components/RoomDropdown/index.tsx]: This is really helpful, thanks for the tips!

File: src/components/RoomDropdown/MemberOptions.tsx [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+    const ref = useRef<HTMLDivElement>(null)
+    const buttonRef = useRef<HTMLDivElement>(null)
+
+    useEffect(() => {
+        function handleClickOutside(event: any) {
+            setDropdownOpen(false)
+        }
+
+        // Bind the event listener
```
Comment (mondoreale) [inline src/components/RoomDropdown/MemberOptions.tsx:149]: This comment looks redundant. The following line is exactly what it says it's doing. Failry obvious.

File: src/icons/CloseIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        width="16"
+        height="16"
+        viewBox="0 0 16 16"
+        fill="none"
+        xmlns="http://www.w3.org/2000/svg"
+    >
+        <path
+            d="M15 1L0.999999 15"
+            stroke="#59799C"
+            stroke-width="1.5"
```
Comment (mondoreale) [inline src/icons/CloseIcon.tsx:12]: `strokeWidth`

File: src/icons/CloseIcon.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+            stroke-linecap="round"
```
Comment (mondoreale) [inline src/icons/CloseIcon.tsx:13]: `strokeLinecap`

File: src/icons/CloseIcon.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        />
+        <path
+            d="M1 1L15 15"
+            stroke="#59799C"
+            stroke-width="1.5"
+            stroke-linecap="round"
```
Comment (mondoreale) [inline src/icons/CloseIcon.tsx:19]: Same.

**Review by mondoreale — APPROVED** [review]
👍🏻

**Comment by mondoreale** [comment]
Getting the following warnings:
```
WARNING in src/components/RoomDropdown/MemberOptions.tsx
  Line 71:7:  'DropDownHeader' is assigned a value but never used  @typescript-eslint/no-unused-vars

src/components/RoomDropdown/index.tsx
  Line 64:7:  'DropDownHeader' is assigned a value but never used  @typescript-eslint/no-unused-vars

src/components/pages/Chat/RoomHeader.tsx
  Line 5:8:  'MoreIcon' is defined but never used  @typescript-eslint/no-unused-vars
```

**Pull request merged**